src/core/filters/packets/bpf/include/packet_filter.h

```diff
@@ -39,6 +42,44 @@ struct retis_packet_filter_ctx {
 
 #define STACK_SIZE		SCRATCH_MEM_START
 
+/* The function defines a placeholder for L2 filters instruction and a
+ * nop frame that will be replaced on load with the actual filtering
+ * instructions. Normally, if no filter gets set, a simple mov r0,
+ * 0x40000 will replace the call. 0x40000 is used as it is also used
+ * by generated cBPF filters, whereas 0 means no match, instead. The
+ * exceeding nops will get removed from the kernel during the load.
+ * If no explicit, nor default filter gets set, call 0xdeadbeef will
+ * fail to load and the verifier will report an error.
+ */
+static __always_inline
+unsigned int packet_filter_mac(struct retis_packet_filter_ctx *ctx)
```

Comment:
Ok, about renaming. I changed it some times until I got tired and decided to flip the coin and stick with `mac_header`, `network_header` naming.

I had the same mixed feelings about the macro. I decided to keep two separate because 2 didn't seem that bad (although, yes, I don't completely love it).